### PR TITLE
Validate enum constant uniqueness for the entire package

### PR DESCRIPTION
**Pull request by swankjesse**

### CODE REVIEW

File: wire-library/wire-schema/src/commonMain/kotlin/com/squareup/wire/schema/ProtoFile.kt [hunk 1]
```diff
@@ -162,10 +162,6 @@ data class ProtoFile(
     return toElement().toSchema()
   }
 
-  fun validate(linker: Linker) {
```
Comment (swankjesse) [inline wire-library/wire-schema/src/commonMain/kotlin/com/squareup/wire/schema/ProtoFile.kt:165]: This was too narrow, they must be unique across all files in the package.

**Review by JakeWharton — APPROVED** [review]
C++, the gift that keeps on giving

**Review by oldergod — COMMENTED** [review]
LGTM

File: gen-tests.gradle.kts [hunk 2]
```diff
@@ -99,8 +99,8 @@ val generateGsonAdapterJavaTests by tasks.creating(JavaExec::class) {
   classpath = wire
   main = "com.squareup.wire.WireCompiler"
   args = listOf(
-          "--proto_path=wire-library/wire-tests/src/commonTest/shared/proto",
           "--proto_path=wire-library/wire-tests/src/commonTest/shared/proto/proto2",
+          "--proto_path=wire-library/wire-tests/src/commonTest/shared/proto/proto3",
```
Comment (swankjesse) [inline gen-tests.gradle.kts:103]: This caused some files to be double-included, which had the unfortunate side-effect of triggering our enums warning. Gotta fix that too!

https://github.com/square/wire/issues/2124

**Pull request merged**